selfhost/typechecker.jakt

```diff
@@ -4184,11 +4184,47 @@ struct Typechecker {
             yield CheckedExpression::QuotedString(val, span)
         }
         Call(call, span) => {
-            yield .typecheck_call(call, caller_scope_id: scope_id, span, this_expr: None, parent_id: None, safety_mode, type_hint, must_be_enum_constructor: false)
+            let maybe_builtin_call = .try_typecheck_builtin_call(call, span, scope_id, safety_mode, type_hint)
```

Comment:
this is the wrong direction as it's making the containing function much larger. If you want to refactor, you'll want to take this logic and move it into its own method, and have that method call into the steps you need.

Reply:
sure. :)

Reply:
I've also removed the optional arguments, it was not adding any value.

Reply:
And did the same thing with the method call.